metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/repositories/ReferenceSequenceRepository.java

```diff
@@ -36,4 +39,6 @@ default void customize(QuerydslBindings bindings, QReferenceSequence referenceSe
                 .first((path, value) -> path.any().equalsIgnoreCase(value.iterator().next()));
     }
 
+    ReferenceSequence findByAccessions(@Param("accessions") List<String> accessions);
```

Comment:
does this query return only one reference-sequence?

Reply:
Discussed this during yesterday's standup; decided to leave as is for now. Natural key for the reference key is going to be the list of its accessions. Because of the way it's implemented, reference sequences which have exactly the same list of accessions will not be duplicated, hence this will always return only one reference sequence.